### PR TITLE
feat(docs): Detail logging and note extra useful options

### PR DESCRIPTION
**Pull request by caksoylar**
Improve recommendations on where to set the logging Kconfig and note a couple extra config that are commonly changed related to logging. I would appreciate a confirmation on my understanding and explanation of these two.

### CODE REVIEW

File: docs/docs/development/usb-logging.md [hunk 1]
```diff
@@ -20,13 +20,17 @@ It is recommended to only enable logging when needed, and not leaving it on by d
...
 
-The `CONFIG_ZMK_USB_LOGGING` KConfig value needs to be set, either by copy and pasting into the `app/prj.conf` file, or by running
-`west build -t menuconfig` and manually enabling the setting in that UI at `ZMK -> Advanced -> USB Logging`.
+The `CONFIG_ZMK_USB_LOGGING` Kconfig enables USB logging. This can be set at the keyboard level, typically in the `config/<your_keyboard>.conf`
+file if you are using a [user config repository](user-setup.md). It can also be enabled at the ZMK level using the `app/prj.conf` file, or other
+search locations described in the [configuration overview](config/index.md#config-file-locations).
+
+Logging can be further configured using Kconfig described in [the Zephyr documentation](https://docs.zephyrproject.org/3.2.0/services/logging/index.html).
+For instance, setting `CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS` to a large value such as `8000` might help catch issues that happen near keyboard
+boot, before you can connect to view the logs. If you frequently get `<err> usb_cdc_acm: Ring buffer full` messages,
```
Comment (xudongzheng) [inline docs/docs/development/usb-logging.md:29]: I don't believe `CONFIG_LOG_BUFFER_SIZE` is related to to the `usb_cdc_acm: Ring buffer full` messages. Also please see https://github.com/zmkfirmware/zmk/issues/1832.

Comment (caksoylar) [reply docs/docs/development/usb-logging.md:29]: Thanks, I think I saw somewhere that it helped with error messages (not `info`) like ` <err> usb_cdc_acm: Ring buffer full, drop 1 bytes`. But noting that could be confusing given #1832 so I'll remove it for now.

**Comment by caksoylar** [comment]
I pared down this PR so that it should be a small but strict improvement, if anyone has any comments I'd appreciate them. Otherwise I can likely merge this in a few days.

**Review by xudongzheng — APPROVED** [review]

**Review by petejohanson — APPROVED** [review]
LGTM.

**Pull request merged**